Update the nuget and microsoft extensions packages

marcpopMSFT opened the pull request:
### Problem
Tooling has identified old packages being downloaded during the build. They aren't used in the SDK itself but we should get clean on CG

marcpopMSFT commented:
@dotnet/source-build I updated to the versions that the SDK is using for these packages. Do I have to add all of them to the version.details.xml now?

mthalman commented:
> @dotnet/source-build I updated to the versions that the SDK is using for these packages. Do I have to add all of them to the version.details.xml now?

The NuGet prebuilts will need to be added to [SBRP](https://github.com/dotnet/source-build-reference-packages) (in the 9.0 branch). The others can be added to the [prebuilt baseline](https://github.com/dotnet/templating/blob/release/9.0.1xx/eng/SourceBuildPrebuiltBaseline.xml).

Forgind commented:
> > @dotnet/source-build I updated to the versions that the SDK is using for these packages. Do I have to add all of them to the version.details.xml now?
> 
> The NuGet prebuilts will need to be added to [SBRP](https://github.com/dotnet/source-build-reference-packages) (in the 9.0 branch). The others can be added to the [prebuilt baseline](https://github.com/dotnet/templating/blob/release/9.0.1xx/eng/SourceBuildPrebuiltBaseline.xml).

Did anyone do this?

marcpopMSFT commented:
@mthalman would adding them to the version.details.xml also work? I went ahead and added nuget here: https://github.com/dotnet/source-build-reference-packages/pull/1216

The rest I should add to the prebuilt exclusion list?

marcpopMSFT commented:
@mthalman ok, updated the SBRP version and added the prebuilt exclusions. Hopefully the source build leg passes. Why wasn't adding the runtime entries in version.details.xml enough to avoid the prebuilt exclusions?

ViktorHofer commented:
@marcpopMSFT this PR seems to have regressed the templating msbuild tasks when being loaded on desktop msbuild. Here's an isolated repro:

_app.csproj_
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <LocalizeTemplates>true</LocalizeTemplates>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.TemplateEngine.Authoring.Tasks" Version="10.0.100-preview.4.25215.8" />
  </ItemGroup>

</Project>
```

_Program.cs_
```C#
// See https://aka.ms/new-console-template for more information
Console.WriteLine("Hello, World!");
```

From a developer command prompt on Windows: `msbuild app.csproj`:

> "C:\temp\templmsbuild\templmsbuild.csproj" (default target) (1:7) ->
(LocalizeTemplatesAfterBuild target) ->
  C:\Users\vihofer\.nuget\packages\microsoft.templateengine.authoring.tasks\10.0.100-preview.4.25215.8\build\Microsoft.
TemplateEngine.Authoring.Tasks.targets(14,5): error MSB4018: The "LocalizeTemplates" task failed unexpectedly. [C:\temp
\templmsbuild\templmsbuild.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.templateengine.authoring.tasks\10.0.100-preview.4.25215.8\build\Microsoft.Te
mplateEngine.Authoring.Tasks.targets(14,5): error MSB4018: System.IO.FileNotFoundException->Microsoft.Build.Framework.B
uildException.GenericBuildTransferredException: Could not load file or assembly 'Microsoft.Extensions.Logging.Abstracti
ons, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot f
ind the file specified. [C:\temp\templmsbuild\templmsbuild.csproj]

ViktorHofer commented:
This is due to libraries in this repo referencing the 9.0.0.0 version of the Microsoft.Extensions.* assemblies but the 9.0.0.3 versioned assemblies being available in the tools folder. That would require binding redirects which a netfx msbuild task can't provide

The overall issue here is that the TFMs in this repo are inconsistent. The [msbuild task](https://github.com/dotnet/templating/blob/b461b060004b8e315c598d46f38abadd0177ed44/tools/Microsoft.TemplateEngine.Authoring.Tasks/Microsoft.TemplateEngine.Authoring.Tasks.csproj#L4) targets net472 but its dependencies target net481 so the netstandard2.0 TFMs are resolved instead which have the wrong assembly version of Microsoft.Extensions* dependencies and would require binding redirects.

I will submit a PR to fix this.

ViktorHofer commented:
Submitted https://github.com/dotnet/templating/pull/8952

ericstj commented:
> They aren't used in the SDK itself but we should get clean on CG

As shown here the `Microsoft.Extensions` packages are used and deployed by the SDK.  I do support this change.

@marcpopMSFT it would be good to add testing to ensure all tasks can load all dependencies on .NETFramework (using msbuild/devenv's bindingRedirects).
